haystack/document_store/sql.py

```diff
@@ -440,6 +440,23 @@ def query_by_embedding(self,
                                   "Change the query type (e.g. by choosing a different retriever) "
                                   "or change the DocumentStore (e.g. to ElasticsearchDocumentStore)")
 
+    def delete_document_by_id(self, id: Union[List[str], str], index: Optional[str] = None):
+        """
+        Delete document given its id
+        :param id: Unique document identifier
+        :param index: Index name to delete the document from.
+        """
+        try:
+            document_to_delete = self.session.query(DocumentORM.id).filter_by(id, index=index)
+            if not id:
+                raise Exception("Document could not be found")
```

Comment:
Possibilities for two times exceptions -
ID not exist -> line 452 raise exception -> line 455 catch it -> line 457 run rollback which will trow exception because not changes do to session.

Reply:
You're right. Should I just raise a generic exception if there's an unexpected database event ? I don't know how to properly this scenario

Reply:
Also this type of delete will not cascade. https://stackoverflow.com/questions/19243964/sqlalchemy-delete-doesnt-cascade

Regarding exception, if handle specific Exception then you can use same approach. Like only handling `SQLAlchemyError` error as this is SQLAlchemy base error. Just do some testing to be sure